.github/workflows/create-build-matrix.py

```diff
@@ -43,10 +43,19 @@ def add_dummy(name: str, artifacts_suffix: str):
     dummy['artifacts-suffix'] = artifacts_suffix
     return dummy
 
-if os.getenv('GITHUB_EVENT_NAME') != 'pull_request':
+enable_package_comparison = os.getenv('enable_package_comparison') == 'true'
+github_event_name = os.getenv('GITHUB_EVENT_NAME')
```

Comment:
Is `os.getenv` case-insensitive? The reason I was wondering is I added the repository variable `ENABLE_PACKAGE_COMPARISON` as upper snake case but then noticed here is lower snake case.

I can see that in the build workflow there is a lower snake case field `enable_package_comparison` definition in the `env` section, which probably is what is used here? However, I was still then left wondering about `GITHUB_EVENT_NAME` which is in upper snake case in the python script, but lower snake case in the `env` section and I couldn't find it defined elsewhere.

Mostly just a clarification to see if I am understanding and following the logic of the action environment correctly.

Reply:
You caused me to think way to critically about my GitHub Actions naming convention habits and I rambled a bit 😅

Short version:

* `os.getenv` case sensitivity is OS-dependent
* Yes it comes from the `env` block and that's why it's lowercase
* `GITHUB_EVENT_NAME` is ambient state, [it's documented here](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)
  * I could switch it to be passed explicitly like `configure-build.py` if you'd like
* The intended conventions here are `UPPER_SNAKE_CASE` for ambient state and `python_naming_conventions`/`MSBuildNamingConventions` for variables passed to the respective system within the scope of the workflow.
* `create-portable-zip.py` didn't follow this convention, whoops. (Will fix with [#1881](https://github.com/bonsai-rx/bonsai/issues/1881))

Long version:

---------------------------------

> Is `os.getenv` case-insensitive?

It is case sensitive on Linux and case-insensitive on Windows. This is the correct case though, this is a weird clash of naming conventions where I used `UPPER_CASE_SNAKE` for the repository variable (which is a common convention and helps make it clear that it's ambient state) but I use the same thing as the Python variable name for passing variables to Python scripts (which is a personal convention as noted below.)

(So I think what my brain is saying is that `ENABLE_PACKAGE_COMPARISON` is ambient state associated with the repository until it is passed to the script after which it no longer is.)

> I can see that in the build workflow there is a lower snake case field `enable_package_comparison` definition in the env section, which probably is what is used here?

Correct, my convention* is that environment variables intended purely for scripts use the convention where they'll be primarily used. So variables intended for Python scripts are `lower_snake_case`, variables for MSBuild are `PascalCase` (EG: `IsReferenceDummyBuild`), etc.

I do this so they look natural where they're used but also it gives a hint that it's something specific to this repo and not ambient environment stuff. It also helps keep things aligned in cases where we don't define the variable. (`ContinuousIntegrationBuild` is an example of this, that's a .NET SDK thing that I did not name.)

Actions workflows are the glue between a lot of different systems so sometimes you get dogs and cats living together.

(*Your comment made me realize that `create-portable-zip.py` doesn't follow this convention though 😅 I might fix that when I address https://github.com/bonsai-rx/bonsai/issues/1881)

> However, I was still then left wondering about `GITHUB_EVENT_NAME` which is in upper snake case in the python script, but lower snake case in the env section and I couldn't find it defined elsewhere.

This one is ambient state from GitHub Actions, it's [documented here](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables).

Using it here was probably just a mood thing. I think originally it was because the script didn't take any environment variables before and I didn't want to add one so I just used the default available thing.

That does make it inconsistent with `configure-build.py` which has it passed manually. Not sure if that's because `configure-build.py` predates GitHub adding `GITHUB_EVENT_NAME` ([it's quite old](https://github.com/MochiLibraries/Biohazrd/blob/4dbaabfb8195853a931a21d9b6468baed9020bf0/.github/workflows/configure-build.py)) or if I wanted it to be explicitly passed so it's clear it influences the outcome of the script. (I suspect the former might be the original reason but the latter would be my reason for keeping it.)

I could change `create-build-matrix.py` to do the same as well for the latter reason/consistency if you'd like.

Reply:
It's fine, no need to change and I agree with all the conventions, thanks for clarifying!